### DISCUSSION TITLE
add highlights above terrain (e.g. using transparent image values)

### DISCUSSION THREAD

**Discussion by wiesehahn** [tylermorganwall/rayshader#179] [Q&A] [answered]
Hi,
I would like to render a scene where certain areas are highlighted with an offset above the terrain. 
The areas are defined by certain values in a raster. I tried to add an RGBA array of the highlight areas as overlay to a NA layer with offset but I cant manage to get it transparent (NA becomes black):
![rayshade_overlay](https://user-images.githubusercontent.com/41429613/129022515-f491d3b4-f47e-425c-9e2f-3725811264ca.png)

It might be related to https://github.com/tylermorganwall/rayshader/discussions/172 and https://github.com/tylermorganwall/rayshader/issues/74.

An alternative solution would be polygonize these areas and add the rendered lines to the scene (similar to https://www.tylermw.com/pathtracing-neon-landscapes-in-r/), but this seems quite complicated and I did not manage to combine the terrain rendering from rayshader and the lines from rayrender. 

Any Ideas?

**Comment by tylermorganwall** [comment, marked as answer]
Check out the latest feature: `render_floating_overlay()`. It should do what you're trying to accomplish here.

https://www.rayshader.com/reference/render_floating_overlay.html
